app/code/Magento/ConfigurableProduct/Model/Plugin/ProductIdentitiesExtender.php

```diff
@@ -46,11 +51,31 @@ public function __construct(Configurable $configurableType, ProductRepositoryInt
      */
     public function afterGetIdentities(Product $subject, array $identities): array
     {
-        foreach ($this->configurableType->getParentIdsByChild($subject->getId()) as $parentId) {
+        if ($subject->getTypeId() !== ConfigurableType::TYPE_CODE) {
+            return $identities;
+        }
+        $parentProductsIdentities = [];
+        foreach ($this->getParentIdsByChild($subject->getId()) as $parentId) {
             $parentProduct = $this->productRepository->getById($parentId);
-            $identities = array_merge($identities, $parentProduct->getIdentities());
+            $parentProductsIdentities[] = $parentProduct->getIdentities();
```

Comment:
Maybe it would be better to rename it to **$parentProductIndentities**

Reply:
It is an array of Products where everyone has Identities